src/main/scala/org/clulab/timenorm/scate/Types.scala

```diff
@@ -842,12 +842,16 @@ case class IntersectionRI(repeatingIntervals: Set[RepeatingInterval],
     if (startInterval.start.plus(1, range).isBefore(ldt)) {
       startPoint = startPoint.plus(1, range)
     }
-    val iterators = sortedRepeatingIntervals.map(_.preceding(startPoint).buffered)
+    var iterators = sortedRepeatingIntervals.map(_.preceding(startPoint).buffered)
 
     Iterator.continually {
-      startPoint = startPoint.minus(1, range)
+      do {
+        startPoint = startPoint.minus(1, range)
+      } while(iterators.exists(startPoint isAfter _.head.start))
       val firstInterval = iterators.head.next
-      val othersAfterStart = iterators.tail.map(it => it.takeWhile(_ => it.head.start isAfter startPoint).toList)
+      val tup = iterators.tail.map(_.span(_.start isAfter startPoint)).unzip
+      val othersAfterStart = tup._1.map(_.toList)
+      iterators = iterators.head :: tup._2.map(_.buffered)
```

Comment:
Maybe slightly prettier:
```
val (othersAfterStart, rest) = iterators.tail.map(_.span(_.start isAfter startPoint)).unzip
iterators = iterators.head :: rest.map(_.buffered)
```
The same goes for below.

Reply:
`othersAfterStart` needs to be `List[List[Interval]]` instead of `List[Iterator[Interval]]`. Otherwise, the  `foldLeft` complains, since the initial value is `List[Interval]`.

Reply:
Right, but you can add the `map` below, before the `foldLeft`.

Reply:
You mean something like `othersAfterStart.map(_.toList).iterator.foldLeft(List(firstInterval))`?